### PR TITLE
Removed cyclic dependency of auth,schema & crud module

### PR DESCRIPTION
**Pull request by sharadregoti**
1) Made auth & schema module independent of each other.
2) Made crud module independent of Auth & Schema.

### CODE REVIEW

File: gateway/modules/crud/internal.go [hunk 1]
```diff
@@ -22,6 +18,11 @@ func (m *Module) InternalCreate(ctx context.Context, dbAlias, project, col strin
 		return err
 	}
 
+	// Validate the create operation
+	if err := helpers.ValidateCreateOperation(ctx, dbAlias, string(crud.GetDBType()), col, m.schemaDoc, req); err != nil {
```
Comment (YourTechBud) [inline gateway/modules/crud/internal.go:22]: Should be the unexported GetDBType

File: gateway/modules/crud/setters.go [hunk 2]
```diff
@@ -119,6 +121,12 @@ func (m *Module) SetSchemaConfig(ctx context.Context, schemas config.DatabaseSch
 		return nil
 	}
 
+	schemaDoc, err := schemaHelpers.Parser(schemas)
```
Comment (YourTechBud) [inline gateway/modules/crud/setters.go:124]: I thought we'll do parsing in modules/operations.go

File: gateway/modules/eventing/crud.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func (m *Module) prepareFindObject(req *model.QueueEventRequest) error {
 
 	// Find the primary keys for the table
 	primaryKeys := make([]string, 0)
-	fields, p := m.schema.GetSchema(dbRequest.DBType, dbRequest.Col)
```
Comment (YourTechBud) [inline gateway/modules/eventing/crud.go:45]: Why can't we store eventing schemas in eventing?

**Review by YourTechBud — CHANGES_REQUESTED** [review]

**Comment by sharadregoti** [comment]
There is no issue created for this PR

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**